engine/flink/components/dev-table/src/main/scala/pl/touk/nussknacker/engine/flink/table/utils/TableTypeConversions.scala

```diff
@@ -15,18 +17,39 @@ object TableTypeConversions {
     )
 
   // not private for test
+  @tailrec
   private[utils] def nuTypeToFlinkTableType(nuType: TypingResult): Option[DataType] = nuType match {
-    case typing.TypedObjectWithValue(underlying, _) => classToFlinkTableType(underlying.klass)
+    case typing.TypedObjectWithValue(typedClass, _) => nuTypeToFlinkTableType(typedClass)
     case typing.TypedClass(klass, _)                => classToFlinkTableType(klass)
-    case typing.TypedNull                           => Some(DataTypes.NULL())
+    case typing.TypedNull                           => Some(DataTypes.NULL)
     case _                                          => None
   }
 
-  // TODO: add remaining conversions
+  // As of Flink 1.19, Flink SQL does not currently support TIMESTAMP WITH TIME ZONE which should get mapped to
+  // java.time.OffsetDateTime even though it appears in Flink docs and source code.
+  // See https://issues.apache.org/jira/browse/FLINK-20869
   private def classToFlinkTableType(klass: Class[_]): Option[DataType] = klass match {
-    case klass if klass == classOf[String]                           => Some(DataTypes.STRING())
-    case klass if klass == classOf[Int] || klass == classOf[Integer] => Some(DataTypes.INT())
-    case _                                                           => None
+    case klass if klass == classOf[String] => Some(DataTypes.STRING)
+
+    case klass if klass == classOf[Boolean] || klass == classOf[java.lang.Boolean] => Some(DataTypes.BOOLEAN)
+
+    case klass if klass == classOf[Byte] || klass == classOf[java.lang.Byte]   => Some(DataTypes.TINYINT)
+    case klass if klass == classOf[Short] || klass == classOf[java.lang.Short] => Some(DataTypes.SMALLINT)
+    case klass if klass == classOf[Int] || klass == classOf[java.lang.Integer] => Some(DataTypes.INT)
+    case klass if klass == classOf[Long] || klass == classOf[java.lang.Long]   => Some(DataTypes.BIGINT)
+
+    case klass if klass == classOf[Float] || klass == classOf[java.lang.Float]   => Some(DataTypes.FLOAT)
+    case klass if klass == classOf[Double] || klass == classOf[java.lang.Double] => Some(DataTypes.DOUBLE)
+    case klass if klass == classOf[java.math.BigDecimal] => Some(DecimalTypeWithDefaultPrecisionAndScale)
+
+    case klass if klass == classOf[java.time.LocalDate]     => Some(DataTypes.DATE)
+    case klass if klass == classOf[java.time.LocalTime]     => Some(DataTypes.TIME)
+    case klass if klass == classOf[java.time.LocalDateTime] => Some(DataTypes.TIMESTAMP)
+    case klass if klass == classOf[java.time.Instant]       => Some(DataTypes.TIMESTAMP_WITH_LOCAL_TIME_ZONE)
+
+    case _ => None
   }
 
+  private val DecimalTypeWithDefaultPrecisionAndScale = DataTypes.DECIMAL(10, 0)
```

Comment:
This is an assumed hardcoded value because we don't have (?) a way to pass information about things like BigDecimals precision and scale in our type system. We could either:
1. (I'm guessing here) Modify our `TypedClass` to carry more information about types
2. Make a 'spacious' default assumption - for example DECIMAL(38,8) so it would be good enough for most purposes. The problem is that when someone uses a decimal with 9 digits after decimal point we don't throw exception, Flink doesn't raise any exceptions, logs - we just get a null value :(